mesonbuild/backend/nonebackend.py

```diff
@@ -0,0 +1,33 @@
+# Copyright 2022 The Meson development team
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+
+import os
+
+from .. import mlog
+from .backends import Backend
+from ..mesonlib import MesonException
+
+class NoneBackend(Backend):
+
+    name = 'none'
+
+    def generate(self):
+        if self.build.get_targets():
+            raise MesonException('None backend cannot generate target rules, try using --backend=ninja')
```

Comment:
That's an interesting point, at least for erroring out early. Pointing to the exact location where the target was added doesn't help much since the only resolution is to use a real backend. :D

Reply:
Thinking about it, it also allow you to write a test case for it:
```meson
testcase expect_error(...)
  library()
endtestcase
```

> Pointing to the exact location where the target was added doesn't help much since the only resolution is to use a real backend. :D

Well, maybe not, a larger project could have optional targets and pointing to where we add targets could help seeing that it's guarded under a `if get_option()`. Maybe overthinking that one :P

Reply:
Ooooh, that's a neat idea now that we have expect_error for internal meson testing.

Reply:
Done.